Relax gettext dependency constraint to allow >= 0.26.0

patrols opened the pull request:
### Summary of changes

Relax gettext dependency constraint to allow >= 0.26.0
This allows users to choose between gettext 0.26.x and 1.0.x versions

[gettext 1.0.0 changelog](https://github.com/elixir-gettext/gettext/blob/main/CHANGELOG.md#v100)

### Checklist

- N/A New functions have typespecs, changed functions were updated
- N/A Same for documentation, including moduledocs
- N/A Tests were added or updated to cover changes
- [x] Commits were squashed into a single coherent commit
- [x] Notes added to CHANGELOG file which describe changes at a high-level

Fixes #782

ddon commented:
Yes, would nice, we also can't update our lib without this

Flo0807 commented on mix.exs:
```diff
@@ -57,7 +57,7 @@ defmodule Timex.Mixfile do
       {:tzdata, "~> 1.1"},
       {:combine, "~> 0.10"},
-      {:gettext, "~> 0.26"},
+      {:gettext, "~> 1.0"},
```
This forces applications to use Gettext v1.0 or higher. As there are no breaking changes and ["very few changes from the latest 0.26 release"](https://github.com/elixir-gettext/gettext/blob/main/CHANGELOG.md), we might want to set the constraint to something like this:

```suggestion
      {:gettext, ">= 0.26.0"},
```

patrols replied:
Yeah, good suggestion. Thanks 👍

patrols committed:
Relax gettext dependency constraint to allow >= 0.26.0

This allows users to choose between gettext 0.26.x and 1.0.x versions,
maintaining backward compatibility while enabling forward compatibility.

tispratik commented:
+1. Can we merge it?

ddon commented:
Would be really nice!

SebastianMueller87 reviewed (commented):
LGTM, can we merge it? 😄

mberrueta-cn commented:
need it too

optikfluffel commented:
@ddon @mberrueta-cn I'm curious why you need this PR so badly. Is there anything preventing you from doing `{:gettext, "~> 1.0", override: true}` as a workaround for now?

SebastianMueller87 commented:
@bitwalker have you already seen this?

lwld commented:
@optikfluffel the answer is in your question: it's a workaround. Not everybody wants to have workarounds in their code. Not every dev might be sure that the `override` is safe in this case.

The question should be "Is there anything preventing this from being merged and released?", not the other way round.

xu-chris reviewed (approved):
Its fine!

xu-chris commented:
@bitwalker It's straightforward PR.

bartekgrzymkowski commented:
Would be great to merge please!
